### PR TITLE
netbsd: secure_getenv compat

### PR DESCRIPTION
**Pull request by carenas**
avoid calling getenv in scenarios where it is not safe (like when
running setuid/setgid)

this should only be effective in NetBSD < 8, as anything that has
MAP_REMAPDUP is using a different implementation of the alloc_chunk
function that doesn't have this deficiency.

### CODE REVIEW

File: sljit_src/sljitProtExecAllocator.c [hunk 1]
```diff
@@ -97,7 +97,24 @@ struct chunk_header {
...
+ * secure_getenv function.
+ * ideally this should never be used, as the standard allocator is
+ * a preferred option for those systems and should be used instead.
+ */
+static SLJIT_INLINE char *secure_getenv(const char *name)
+{
+	int secure_execution_required = getuid() != geteuid()
+					|| getgid() != getegid();
+
+	return secure_execution_required ? NULL : getenv(name);
```
Comment (zherczeg) [inline sljit_src/sljitProtExecAllocator.c:113]: This is ok, but what about making it a define?
```
#define secure_getenv(name) \
  (cond) ? NULL : getenv(name);
```

**Comment by krytarowski** [comment]
If this is for NetBSD<8 then is it worth the patch? NetBSD-7 will go EOL soon.

**Comment by zherczeg** [comment]
I am ok with the patch, but I don't know how netbsd maintenace works. Most OSes only do security updates after a release, so they never use the latest code for an old revision. Users can do anything though.

**Comment by carenas** [comment]
the original code introduces a security issue that this fixes, tragically it is included in pkg_src so as such affects currently all NetBSD versions.

the comment about NetBSD < 8 not being affected also requires that the sljitProtExecAllocator be patched with 9a3d3998dcaf151f5c562f8dde457790fe0d167c, that was merged first but hasn't been released yet

**Pull request merged**